roachtest: fix the DROP test

tbg opened the pull request:
We were unable to delete from the stock table which was referenced from the
order_line table. Just drop the constraint.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/23855)
<!-- Reviewable:end -->

tbg committed:
roachtest: fix the DROP test

We were unable to delete from the stock table which was referenced from the
order_line table. Just drop the constraint.

Release note: None

benesch reviewed (approved):
🎉 thanks Tobi!

tschottdorf commented:
TFTR!
